File: ios/Classes/PrintQrCodeModel.m
#import "PrintQrCodeModel.h"

@implementation PrintQrCodeModel

+ (PrintQrCodeModel *)fromJson:(NSString *)json {
    PrintQrCodeModel *model = [[PrintQrCodeModel alloc] init];
    NSData *jsonData = [json dataUsingEncoding:NSUTF8StringEncoding];
    NSError *error = nil;
    NSDictionary *dictionary = [NSJSONSerialization JSONObjectWithData:jsonData
                                                               options:0
                                                                 error:&error];
    if (error != nil || ![dictionary isKindOfClass:[NSDictionary class]]) {
        model.data = @"";
        model.size = 0;
        return model;
    }
    model.data = [dictionary[@"data"] isKindOfClass:[NSString class]]
        ? dictionary[@"data"]
        : @"";
    model.size = [dictionary[@"size"] doubleValue];
    return model;
}

@end

File: android/src/main/java/com/zh4dev/niimbot_print/Model/PrintQrCodeModel.java
package com.zh4dev.niimbot_print.Model;

import com.google.gson.Gson;

public class PrintQrCodeModel {

    private String data;
    private double size;

    public String getData() {
        return data;
    }

    public double getSize() {
        return size;
    }

    public static PrintQrCodeModel fromJson(String json) {
        return new Gson().fromJson(json, PrintQrCodeModel.class);
    }
}
